x-pack/plugins/files/server/file_client/file_client.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { FileKind } from '../../common/types';
+import type { FileMetadataClient } from './file_metadata_client';
+import type { BlobStorageClient } from '../blob_storage_service';
+import { enforceMaxByteSizeTransform } from './stream_transforms';
+
+/**
+ * Wraps the {@link FileMetadataClient} and {@link BlobStorageClient} client
+ * to provide basic file CRUD functionality.
+ */
+export class FileClient {
+  constructor(
+    private fileKindDescriptor: FileKind,
+    private readonly metadataClient: FileMetadataClient,
+    private readonly blobStorageClient: BlobStorageClient
+  ) {}
+
+  public get fileKind(): string {
+    return this.fileKindDescriptor.id;
+  }
+
+  public create: FileMetadataClient['create'] = async (arg) => {
+    return this.metadataClient.create(arg);
+  };
+
+  public get: FileMetadataClient['get'] = async (arg) => {
+    return this.metadataClient.get(arg);
+  };
+
+  public update: FileMetadataClient['update'] = (arg) => {
```

Comment:
Something to think about: @sixstringcode mentioned that security team has a use case where text files are edited in text editor. If we would allow file editing, initial idea that comes to mind: we could try to leave file SO as-is, but re-upload the blob.